google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionContextFutureImpl.java

```diff
@@ -109,6 +109,7 @@ public Timestamp get(long timeout, TimeUnit unit)
             @Override
             public void onFailure(Throwable t) {
               mgr.onError(t);
+              statementResult.setException(t);
```

Comment:
This missing statement was what could cause the infinite wait. The result of this specific statement would never be done if the previous statement (`input`) would fail.